Problem with @testing-library/jest-dom types after updating to v0.30.0

### Describe the bug

`vitest.config.ts`:

```ts
export default mergeConfig(
  viteConfig,
  defineConfig({
    test: {
      setupFiles: './scripts/setup-tests.ts',
    },
    // ....
  }),
)
```

`./scripts/setup-tests.ts`:

```
import matchers from '@testing-library/jest-dom/matchers'
import { expect } from 'vitest'

expect.extend(matchers)
```


`tsconfig.json`:

```json
{
  "compilerOptions": {
    "types": ["node", "@testing-library/jest-dom"],
  // ...
  }
}
```

I am getting typescript errors:

```
components/button/index.test.tsx:67:38 - error TS2339: Property 'toHaveClass' does not exist on type 'Assertion<HTMLElement>'.

67   expect(getByText(trimmedText)).not.toHaveClass('full')
                                        ~~~~~~~~~~~

components/button/index.test.tsx:68:31 - error TS2339: Property 'toHaveClass' does not exist on type 'Assertion<HTMLElement>'.

68   expect(getByText(fullText)).toHaveClass('full')
                                 ~~~~~~~~~~~

components/button/index.test.tsx:105:36 - error TS2339: Property 'toBeInTheDocument' does not exist on type 'Assertion<HTMLElement>'.

105   expect(screen.getByRole('link')).toBeInTheDocument()
```

### System Info

```shell
System:
    OS: macOS 13.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 413.16 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.asdf/installs/nodejs/18.15.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.5.0 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 112.0.5615.49
    Firefox: 111.0.1
    Safari: 16.4
  npmPackages:
    @vitejs/plugin-react: ^3.1.0 => 3.1.0
    @vitest/coverage-c8: ^0.30.0 => 0.30.0
    vite: ^4.2.1 => 4.2.1
    vitest: ^0.30.0 => 0.30.0
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vitest/src/types/global.ts
+++ b/packages/vitest/src/types/global.ts
@@ -44,7 +44,8 @@ declare global {
   // support augmenting jest.Matchers by other libraries
   namespace jest {
 
-    interface Matchers<_R, _T = {}> {}
+    // eslint-disable-next-line unused-imports/no-unused-vars
+    interface Matchers<R, T = {}> {}
   }
 
   namespace Vi {

--- a/test/core/test/jest-expect.test.ts
+++ b/test/core/test/jest-expect.test.ts
@@ -12,7 +12,14 @@ interface CustomMatchers<R = unknown> {
   toBeTestedSync(): R
   toBeTestedPromise(): R
 }
+
 declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toBeJestCompatible(): R
+    }
+  }
+
   namespace Vi {
     interface JestAssertion extends CustomMatchers {}
     interface AsymmetricMatchersContaining extends CustomMatchers {}
@@ -182,6 +189,12 @@ describe('jest-expect', () => {
           message: () => 'toBeTestedPromise',
         })
       },
+      toBeJestCompatible() {
+        return {
+          pass: true,
+          message: () => '',
+        }
+      },
     })
 
     expect(5).toBeDividedBy(5)
@@ -195,6 +208,8 @@ describe('jest-expect', () => {
     expect(() => expect(null).toBeTestedSync()).toThrowError('toBeTestedSync')
     await expect(async () => await expect(null).toBeTestedAsync()).rejects.toThrowError('toBeTestedAsync')
     await expect(async () => await expect(null).toBeTestedPromise()).rejects.toThrowError('toBeTestedPromise')
+
+    expect(expect).toBeJestCompatible()
   })
 
   it('object', () => {

